### DISCUSSION TITLE
[Theory or Implementation] The relative motion between two bodies are not showing

### DISCUSSION THREAD

**Discussion by chakramtalla** [WEC-Sim/WEC-Sim#1334] [Q&A] [answered]
I am facing an issue that the power is not generated 

I am using RM3 type model , point absorber for 4m buoy cases

![image](https://github.com/user-attachments/assets/954ede5e-9bdb-40c4-bcf0-798a9d19f29d)

and also, I tried to save my files in STL format. if I am using these files in wecsim the simulations were showing like this
![image](https://github.com/user-attachments/assets/ddbad5c5-6eb9-4d52-a9c5-17798eb8fcfa)

kindly rectify my issues , where I am getting trouble

**Comment by MShabara** [comment, marked as answer]
Hi @chakramtalla 

I think your problem is not a WEC-Sim source code issue, its an issue with your model which can be moved to the discussions tab.
 
For your first question, I guess you are using a passive damping PTO, and you have a missing negative sign in your power equations.  

For your second question, please check Nathan's response to issue #1088

**Reply by chakramtalla** [reply]
Thank you for  reply. I wasn't able to find the relative motion between float and plate; I felt that my power was not generated due to this, but when I checked, the RAOs were perfectly fine. can you suggest what exactly happen between two bodies. Do we need to give any constraint

**Reply by MShabara** [reply]
I’m sorry, but I didn’t fully understand your question. Could you please provide more details about the problem you’re experiencing? It would help if you could clarify specifics of the issue. Also, please keep in mind that I’m not familiar with your code or the specific model setup, so any additional context would be helpful for me to assist you better.

**Reply by chakramtalla** [reply]
"I transferred the files from ANSYS AQWA and generated the `.h5` file, then checked for RAOs. After that, I ran the WEC-Sim model similar to the RM3 case. The simulation ran fine, but when I checked for power generated due to the actuation force, it showed no results. I'm working with buoy sizes of 2m and 3m, similar to the RM3 model. Can you help identify the possible issues causing this?"

**Reply by MShabara** [reply]
I do not understand what you mean by "no results"; in your first comment you showed a power curve for the net power lost in the PTO.

Please use this line of code in the `underDefinedFunctions.m` related to your model to plot the generated power.
`plot(output.ptos.time,-output.ptos.powerInternalMechanics(:,3))`
